common/components/QueueItem.tsx

```diff
@@ -142,49 +61,71 @@ export async function QueueItem({
                       d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
                     />
                   </svg>
-                </div>
-              );
+                </div>)
           })}
         </button>
       );
     } else if (type === "search") {
       return (
         <button
           onclick={async () => {
-            await handleButtonClick();
-            await sleep(250);
-            const session = await getSessionWithCode(code);
-            if (!session) return;
+            const clickCode = code.length ? code : code.val
+            const session = await getSessionWithCode(clickCode);
+            // check if session is null
+            if (!session) {
+              return;
+            }
 
-            if ((session.queue.some((v) => v.id == item.id) || session.currentlyPlaying?.id == item.id)) {
+            // check if queue already contains the video
+            if (session.queue.some((v) => v.id == item.id)) {
               return;
             }
+
             session?.$.queue.val.push(item);
-            toggleLike(code, item.id);
+
+            const recommendations = await getRecommendations(session?.$.queue.val);
```

Comment:
maybe we should move all of this into a function on the backend like `updateRecommendations`?